Source/MQTTnet/Client/MqttClient.cs

```diff
@@ -38,8 +38,11 @@ public class MqttClient : Disposable, IMqttClient
 
         IMqttChannelAdapter _adapter;
         bool _cleanDisconnectInitiated;
-        long _isDisconnectPending;
-        bool _isConnected;
+        volatile int _connectState;
+        const int CONNECT_STATED_DISCONNECTED = 0;
```

Comment:
Please convert this to an enum with PascalCase style because this common in C#. You can also add a private method as a replacement of the _Interlocked.CompareExchange_ calls which will also accept these enums only. Internally the new method will cast the num as integers to that we use the enum only in business code and have the technical conversion and thread safe access stuff separated.